prov/sockets/src/sock_progress.c

```diff
@@ -2466,6 +2466,8 @@ static int sock_pe_progress_rx_ep(struct sock_pe *pe, struct sock_ep_attr *ep_at
 	fastlock_acquire(&map->lock);
 	for (i = 0; i < num_fds; i++) {
 		fd = sock_epoll_get_fd_at_index(&map->epoll_set, i);
+		if(fd == -1) /* failed to lookup fd due to connection failures */
+			continue;
```

Comment:
as i can see -1 is returned on windows platform only (there is bit different socket state handling)